extract_unique_sensors_01.py

```diff
@@ -48,7 +56,8 @@ def read_sensors(file_name, url):
             sensors_url = urlopen(url)
             sensors_data = sensors_url.readlines()
         except Exception:                         
-            print("Failed to retrieve sensor data")
+            logging.error(
```

Comment:
Instead of enabling the exception you could instead use, `exception` loglevel

See: https://docs.python.org/3/library/logging.html#logging.Logger.exception